components/service-waiter/cmd/component.go

```diff
@@ -127,8 +142,69 @@ func init() {
 	componentCmd.Flags().StringVar(&componentCmdOpt.namespace, "namespace", "", "The namespace of deployment")
 	componentCmd.Flags().StringVar(&componentCmdOpt.component, "component", "", "Component name of deployment")
 	componentCmd.Flags().StringVar(&componentCmdOpt.labels, "labels", "", "Labels of deployment")
+	componentCmd.Flags().StringVar(&componentCmdOpt.ideMetricsHost, "ide-metrics-host", "", "Host of ide metrics")
+	componentCmd.Flags().DurationVar(&componentCmdOpt.featureFlagTimeout, "feature-flag-timeout", 3*time.Minute, "The maximum time to wait for feature flag")
+	componentCmd.Flags().StringVar(&componentCmdOpt.gitpodHost, "gitpod-host", "", "Domain of Gitpod installation")
 
 	_ = componentCmd.MarkFlagRequired("namespace")
 	_ = componentCmd.MarkFlagRequired("component")
 	_ = componentCmd.MarkFlagRequired("labels")
 }
+
+func startWaitFeatureFlag(ctx context.Context, timeout time.Duration) {
+	featureFlagCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	client := experiments.NewClient(experiments.WithDefaultClient(nil), experiments.WithPollInterval(time.Second*3))
+	defaultSkip := true
+	if client == nil {
+		log.Error("failed to create experiments client, skip immediately")
+		shouldSkipComponentWaiter = defaultSkip
+		metrics.AddSkipComponentsCounter(componentCmdOpt.ideMetricsHost, strconv.FormatBool(shouldSkipComponentWaiter), false)
+		return
+	}
+	getShouldSkipComponentWaiter := func() {
+		startTime := time.Now()
+		value, isActualValue, fetchTimes := ActualWaitFeatureFlag(featureFlagCtx, client, defaultSkip)
+		avgTime := time.Since(startTime) / time.Duration(fetchTimes)
```

Comment:
@mustard-mh When waiting for longer than timeout, I got a integer divide by zero here (I scaled server down to 0 replicas to test):
```
panic: runtime error: integer divide by zero

goroutine 18 [running]:
github.com/gitpod-io/gitpod/service-waiter/cmd.startWaitFeatureFlag.func1()
        github.com/gitpod-io/gitpod/service-waiter/cmd/component.go:168 +0x4e5
github.com/gitpod-io/gitpod/service-waiter/cmd.startWaitFeatureFlag({0x1a40810?, 0xc000434c40?}, 0x0?)
        github.com/gitpod-io/gitpod/service-waiter/cmd/component.go:174 +0x1be
created by github.com/gitpod-io/gitpod/service-waiter/cmd.glob..func2 in goroutine 1
        github.com/gitpod-io/gitpod/service-waiter/cmd/component.go:56 +0x45b
```